src/Mollie/WC/Helper/ApplePayDirectHandler.php

```diff
@@ -0,0 +1,1180 @@
+            function () {
+                $this->applePayDirectButton();
+            }
+        );
+        admin_url('admin-ajax.php');
+        $this->bootstrapAjaxRequest();
+    }
+
+    /**
+     * Checks if the server is HTTPS
+     *
+     * @return bool
+     */
+    private function isApplePayCompatible()
+    {
+        return !empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] != 'off';
+    }
+
+    /**
+     * Sets the appropriate data to send to ApplePay script
+     * Data differs between product page and cart page
+     *
+     * @return array|bool
+     */
+    public function applePayScriptData()
+    {
+        $base_location = wc_get_base_location();
+        $shopCountryCode = $base_location['country'];
+        $currencyCode = get_woocommerce_currency();
+        $totalLabel = get_bloginfo('name');
+        if (is_product()) {
+            $product = wc_get_product(get_the_id());
+            if (!$product) {
+                return false;
+            }
+            $isVariation = false;
+            if ($product->get_type() === 'variable') {
+                $isVariation = true;
+            }
+            $productNeedShipping = $this->checkIfNeedShipping($product);
+            $productId = get_the_id();
+            $productPrice = $product->get_price();
+
+            return [
+                'product' => [
+                    'needShipping' => $productNeedShipping,
+                    'id' => $productId,
+                    'price' => $productPrice,
+                    'isVariation' => $isVariation,
+                ],
+                'shop' => [
+                    'countryCode' => $shopCountryCode,
+                    'currencyCode' => $currencyCode,
+                    'totalLabel' => $totalLabel
+                ],
+                'ajaxUrl' => admin_url('admin-ajax.php')
+            ];
+        }
+        if (is_cart()) {
+            $cart = WC()->cart;
+
+            return [
+                'product' => [
+                    'needShipping' => $cart->needs_shipping(),
+                    'subtotal' => $cart->get_subtotal(),
+
+                ],
+                'shop' => [
+                    'countryCode' => $shopCountryCode,
+                    'currencyCode' => $currencyCode,
+                    'totalLabel' => $totalLabel
+                ],
+                'ajaxUrl' => admin_url('admin-ajax.php')
+            ];
+        }
+        return [];
+    }
+
+    /**
+     * Method to validate the merchant against Apple system through Mollie
+     * On fail triggers and option that shows an admin notice showing the error
+     * On success returns the validation data to the script
+     */
+    public function validateMerchant()
+    {
+        $applePayRequestDataObject = $this->applePayDataObjectHttp();
+        $applePayRequestDataObject->validationData($_POST);
+
+        if (!$this->isNonceValid($applePayRequestDataObject)) {
+            wp_send_json_error('no nonce');
```

Comment:
Not sure about this, what if in the client you use messages with error as messages for the end user? You'll show `no nonce` 🤔  I would just return.